synapse/lib/editor.py

```diff
@@ -0,0 +1,998 @@
+import asyncio
+import logging
+import collections
+
+import synapse.exc as s_exc
+import synapse.common as s_common
+
+import synapse.lib.chop as s_chop
+import synapse.lib.node as s_node
+import synapse.lib.time as s_time
+import synapse.lib.layer as s_layer
+import synapse.lib.scope as s_scope
+import synapse.lib.types as s_types
+
+logger = logging.getLogger(__name__)
+
+class ProtoNode(s_node.NodeBase):
+    '''
+    A prototype node used for staging node adds using a NodeEditor.
+    '''
+    def __init__(self, ctx, buid, form, valu, node):
+        self.ctx = ctx
```

Comment:
If `ctx` is the editor, perhaps it would be best to just name this `editor` for clarity?